addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -1926,18 +1941,32 @@ export class PosStore extends WithLazyGetterTrap {
             if (preset.identification === "name" && !order.floating_order_name && !order.table_id) {
                 order.floating_order_name = order.getPartner()?.name;
                 if (!order.floating_order_name) {
-                    this.editFloatingOrderName(order);
+                    await this.editFloatingOrderName(order);
                 }
+                order = this.getOrder();
```

Comment:
Put this right after `await this.editFloatingOrderName(order);`. Also leave a comment why you need to re-set the value of the `order` variable.